core/modules/parser/SelectParser.cc

```diff
@@ -366,6 +366,13 @@ SelectParser::Ptr SelectParser::newInstance(std::string const& statement, AntlrV
     return std::shared_ptr<SelectParser>(new SelectParser(statement, v));
 }
 
+std::shared_ptr<query::SelectStmt> SelectParser::makeSelectStmt(
+        std::string const& statement, AntlrVersion v) {
+    auto parser = newInstance(statement, v);
+    parser->setup();
```

Comment:
Should the setup call be done in newInstance()?

Reply:
no, because that makes a new instance of the parser, this makes a SelectStmt, which requires calling setup on the parser. It's a little odd, I know. I don't really like the API of the SelectParser, requiring the call to setup. I haven't wanted to go ahead and refactor it yet, but I may get there at some point. It'll be more likely after we drop the antlr 2 parser entirely.